pkgs/servers/pufferpanel/default.nix

```diff
@@ -2,23 +2,54 @@
 , buildGoModule
 , fetchFromGitHub
 , makeWrapper
-, pkgs
 , stdenv
 , fetchzip
-, jdk
-, nodejs
+, fetchpatch
 , pathDeps ? [ ]
 }:
 
 buildGoModule rec {
   pname = "pufferpanel";
-  version = "2.2.0";
+  version = "2.6.6";
+
+  patches = [
+    # Bump go-sqlite3 version.
+    # See https://github.com/PufferPanel/PufferPanel/pull/1240
+    (fetchpatch {
+      url = "https://github.com/PufferPanel/PufferPanel/pull/1240/commits/3065dca2d9b05a56789971ccf0f43a7079a390b8.patch";
+      hash = "sha256-ygMrhJoba8swoRBBii7BEiLihqOebLUtSH7os7W3s+k=";
+    })
+
+    # Fix errors in tests.
+    # See https://github.com/PufferPanel/PufferPanel/pull/1241
+    (fetchpatch {
+      url = "https://github.com/PufferPanel/PufferPanel/pull/1241/commits/ffd21bce4bff3040c8e3e783e5b4779222e7a3a5.patch";
+      hash = "sha256-BzGfcWhzRrCHKkAhWf0uvXiiiutWqthn/ed7bN2hR8U=";
+    })
```

Comment:
1241 is a minor change that fixes package check phase (running `go test`). There is one test that simply does not compile (likely after some refactoring in PufferPanel), and another with the wrong input (casting `[]T` to `[]string` succeeds if `T` can be cast to `string`, and that’s the case with test data; this patch replaces it with `map` value from the test data).

1240 bumps sqlite3 library so that we have a fix for a false positive from compiler (see https://github.com/mattn/go-sqlite3/issues/822 and https://github.com/mattn/go-sqlite3/issues/803#issuecomment-829462682). I guess we can use `-Wno-return-local-addr`? There are two packages in nixpkgs that do this,
https://github.com/NixOS/nixpkgs/blob/c7a94612a7a5b26dba69cdb87381ef9359d11d93/pkgs/servers/photoprism/backend.nix#L26-L27 https://github.com/NixOS/nixpkgs/blob/c7a94612a7a5b26dba69cdb87381ef9359d11d93/pkgs/servers/misc/navidrome/default.nix#L49

I’m a bit hesitant to do that since [go-sqlite3 v2 release is an accident](https://github.com/mattn/go-sqlite3/blob/85a15a7254f28c81069dbbc51201e5025b5a74bc/README.md?plain=1#L10-L12) that is almost 3 years old and ships with sqlite 3.32.2 while v1.14.16 ships with 3.39.4. Note that it should also be possible to define `USE_LIBSQLITE3` and pass sqlite package as build input, but currently there are no such cases in nixpkgs.

Reply:
Thank you for the explanation!

I think a little:
`     # Bump go-sqlite3 version to avoid a C compiler error`
…for 1240 would clarify that one, and I think 1241 was a false positive from me reviewing it, sorry :P

Reply:
Done.